src/test/java/org/minions/devfund/sergio/SquarelotronTest.java

```diff
@@ -0,0 +1,400 @@
+     */
+    @Test
+    public void testMainDiagonalFlipIn2x2SquareRingOne() {
+        final int size = 2;
+        squarelotron = new Squarelotron(size);
+        test.append("01 03").append("\n")
+                .append("02 04");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.mainDiagonalFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testMainDiagonalFlipIn6x6SquareRingOne() {
+        final int size = 6;
+        squarelotron = new Squarelotron(size);
+        test.append("01 07 13 19 25 31").append("\n")
+                .append("02 08 09 10 11 32").append("\n")
+                .append("03 14 15 16 17 33").append("\n")
+                .append("04 20 21 22 23 34").append("\n")
+                .append("05 26 27 28 29 35").append("\n")
+                .append("06 12 18 24 30 36");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.mainDiagonalFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testMainDiagonalFlipIn4x4RingTwo() {
+        final int size = 4;
+        squarelotron = new Squarelotron(size);
+        test.append("01 02 03 04").append("\n")
+                .append("05 06 10 08").append("\n")
+                .append("09 07 11 12").append("\n")
+                .append("13 14 15 16");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.mainDiagonalFlip(2)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values of the processed square of the method mainDiagonalFlip.
+     */
+    @Test
+    public void testMainDiagonalFlipOverCharge() {
+        final int size = 4;
+        squarelotron = new Squarelotron(size);
+        test.append("01 05 09 13").append("\n")
+                .append("02 06 07 14").append("\n")
+                .append("03 10 11 15").append("\n")
+                .append("04 08 12 16");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.mainDiagonalFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 2x2.
+     */
+    @Test
+    public void testRotateEastIn2x2OneTurn() {
+        final int size = 2;
+        final int turns = 1;
+        squarelotron = new Squarelotron(size);
+        test.append("03 01").append("\n")
+                .append("04 02");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.rotateEast(turns)));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testRotateEastIn4x4TwoTurns() {
+        final int size = 4;
+        final int turns = 2;
+        squarelotron = new Squarelotron(size);
+        test.append("16 15 14 13").append("\n")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "16 15 14 13" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)